server/webapp/WEB-INF/rails/webpack/views/pages/pipelines/spec/template_editor_spec.tsx

```diff
@@ -73,6 +89,18 @@ describe("AddPipeline: TemplateEditor", () => {
     const dropdown = helper.byTestId("form-field-input-template");
     expect(dropdown).not.toExist();
   });
+
+  it("should populate parameters when present", () => {
+    spyOn(TemplateConfig, "getTemplate").and.callFake((name, onSuccess) => {
+      onSuccess(new TemplateConfig(name, [new PipelineParameter("paramName", "")]));
+      expect(paramList().length).toBe(1);
```

Comment:
Putting expectations here isn't quite enough because if the `onSuccess()` handler never runs, the test won't fail. Even though this function is no longer asynchronous because of the stubbing, it could be valuable to make the test asynchronous and call `done()` within the `onSuccess()` to validate that it runs at all.

I did check out the code and try this, and I found the callback does not actually run :)

Reply:
@kierarad it's probably never getting there because the component is mounted with `new EmptyCache()`

Reply:
These issues should now be fixed in d5fa251